Module#04/repository-spring-jpa/src/main/java/com/epam/esm/repository/impl/UserRepoImpl.java

```diff
@@ -0,0 +1,77 @@
+package com.epam.esm.repository.impl;
+
+import com.epam.esm.dao.UserDao;
+import com.epam.esm.entity.User;
+import com.epam.esm.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.stream.Collectors.toCollection;
+
+@RequiredArgsConstructor
+public class UserRepoImpl implements UserDao {
+
+    private final UserRepository userRepository;
+
+    /**
+     * Retrieves all entities.
+     *
+     * @param pageable the pagination information
+     * @return a list of entities
+     */
+    @Override
+    public List<User> getAllBy(final Pageable pageable) {
+        return userRepository.findAll(pageable)
```

Comment:
if I am not wrong findAll() method returns ArrayList by default

Reply:
> if I am not wrong findAll() method returns ArrayList by default

> PagingAndSortingRepository returns a Page of entities meeting the paging restriction provided in the Pageable object.